### DISCUSSION TITLE
Dispatcher Not Configured For Router Path

### DISCUSSION THREAD

**Discussion by kstatz12** [akkadotnet/akka.net#6056] [General]
I have this utility to make creating non homogeneous broadcast groups
```
public class RouterBuilder
{
    private readonly List<Func<IActorRef>> pathsFns;

    public RouterBuilder() => this.pathsFns = new();

    public RouterBuilder Add(Func<IActorRef> fn)
    {
        this.pathsFns.Add(fn);
        return this;
    }

    public IActorRef BroadcastGroup(IActorContext ctx, string name) =>
        ctx.ActorOf(Props.Empty.WithRouter(new BroadcastGroup(this.pathsFns.Select(x => x().Path.ToString()), name)));
}

``` 

which is used as such:

```
public DomainEventHandlerActor()
        {
            var rb = new RouterBuilder()
                .Add(() => Context.ActorOf(Context.System.DI().Props<UserCacheActor>()))
                .Add(() => Context.ActorOf(Context.System.DI().Props<LoginCacheActor>()))
                .Add(() => Context.ActorOf(Context.System.DI().Props<GlobalPublishActor>()));


            var group = rb.BroadcastGroup(Context, "user_actor_group");

            ReceiveAny(e =>
            {
                group.Tell(e, Self);
            });
        }
```

but when i spin up the project I get this error in the logs
```Dispatcher [user_actor_group] not configured for router of path [akka://CerosActorSystem/user/$b/$a/$d]\n```

**Comment by Arkatufus** [comment]
Are you sure that you have a router dispatcher with id "user_actor_group", or are you intending on naming the new actor "user_actor_group"?
The constructor `BroadcastGroup(IEnumerable<string>, string)` is ment to pass in a dispatcher id name into the new `BroadcastGroup` instance, not giving it a name.
Maybe you intended to write `ctx.ActorOf(Props.Empty.WithRouter(new BroadcastGroup(this.pathsFns.Select(x => x().Path.ToString()))), name);` instead?

**Reply by kstatz12** [reply]
Im a big ol dummy, thanks! i got lost in the parens.
